src/Aspire.Hosting.Azure.Provisioning/Provisioners/BicepProvisioner.cs

```diff
@@ -147,6 +147,7 @@ await foreach (var kv in keyVaults.GetAllAsync(cancellationToken: cancellationTo
                     int i => i,
                     bool b => b,
                     JsonNode node => node,
+                    BicepOutputReference reference => reference.Value,
```

Comment:
Making this work would mean restructuring this based on dependencies. I don't really want to do that in this PR, so maybe I should throw for now? 

The approach I would take here is generating a big bicep module so that I submit a single deployment instead of doing N deployments. That would allow the bicep/arm to figure out the ordering, but it's a non-trivial amount of work, but doable.